aspnet/publishing/linuxproduction.rst

```diff
@@ -0,0 +1,190 @@
+
+.. code-block:: nginx
+   :emphasize-lines: 7
+
+    server {
+        listen 80;
+        location / {
+            proxy_pass http://unix:/var/aspnet/HelloMVC/kestrel.sock;
+            proxy_http_version 1.1;
+            proxy_set_header Upgrade $http_upgrade;
+            proxy_set_header Connection keep-alive;
+            proxy_set_header Host $host;
+            proxy_cache_bypass $http_upgrade;
+        }
+    }
+
+This is one of the simplest configuration files for Nginx that forwards incoming public traffic on your port ``80`` to a unix socket that your web application will listen on. You can specify this Unix socket in your ``project.json`` file.
+
+.. code-block:: json
+    :caption: project.json (truncated)
+
+    "commands": {
+        "web": "Microsoft.AspNet.Server.Kestrel --server.urls http://unix:/var/aspnet/HelloMVC/kestrel.sock",
+    },
+
+.. note::
+
+    The ``proxy_set_header Connection keep-alive;`` is **required** as a temporary workaround a `known bug in Kestrel <https://github.com/aspnet/KestrelHttpServer/issues/341>`_.
+
+You might want to look at ``etc/nginx/nginx.conf`` to make other necessary changes to your nginx environment to configure your nginx environment.
+
+Once you have completed making changes to your nginx configuration you can run ``sudo nginx -t`` to verify the syntax of your configuration files. If you see any no errors you can ask nginx to pick up the changes by running ``sudo nginx -s reload``.
+
+
+
+Monitoring our Web Application
+------------------------------
+
+Nginx will forward requests to your Kestrel server, however unlike IIS on Windows, it does not mangage your Kestrel process. In this tutorial, we will use `supervisor <http://supervisord.org/>`_ to start our application on system boot and restart our process in the event of a failure.
+
+
+Installing supervisor
+~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: bash
+
+    sudo apt-get install supervisor
+
+.. note::
+
+    ``supervisor`` is a python based tool and you can acquire it through `pip <http://supervisord.org/installing.html#installing-via-pip>`_ or `easy_install <http://supervisord.org/installing.html#internet-installing-with-setuptools>`_ instead.
+
+
+Configuring supervisor
+~~~~~~~~~~~~~~~~~~~~~~
+
+Supervisor works by creating child processes based on data in its configuration file. When a child process dies, supervisor is notified via the ``SIGCHILD`` signal and supervisor can react accordingly and restart your web application.
+
+To have supervisor monitor our application, we will add a file to the ``/etc/supervisor/conf.d/`` directory.
+
+.. code-block:: ini
+    :caption: /etc/supervisor/conf.d/hellomvc.conf
+
+    [program:hellomvc]
+    command=bash /var/aspnet/HelloMVC/approot/web
+    autostart=true
+    autorestart=true
+    stderr_logfile=/var/log/hellomvc.err.log
+    stdout_logfile=/var/log/hellomvc.out.log
+    environment=ASPNET_ENV=Production
+    user=aspnet
```

Comment:
Either use the default `www-data` user (which I'd recommend), or explicitly show the commands to create a user called `aspnet` :)